pkg/utils/nodeutils.go

```diff
@@ -152,11 +152,16 @@ func AreKubeletConfigsRendered(pool *mcfgv1.MachineConfigPool, client runtimecli
 		return false, fmt.Errorf("failed to unmarshal machine config %s: %w", currentKCMCName, err), diffString
 	}
 
-	encodedKC, err := jsonpath.Get("storage.files[0].contents.source", obj)
+	// Filter to kubelet.conf file as other files (e.g. /etc/node-sizing-enabled.env) can exist.
+	encodedKC, err := jsonpath.Get(`$.storage.files[?(@.path == "/etc/kubernetes/kubelet.conf")].contents.source`, obj)
 	if err != nil {
 		return false, fmt.Errorf("failed to get encoded kubelet config from machine config %s: %w", currentKCMCName, err), diffString
 	}
-	encodedKCStr := encodedKC.(string)
+	encodedKCSlice := encodedKC.([]interface{})
+	if len(encodedKCSlice) == 0 {
+		return false, fmt.Errorf("encoded kubeletconfig %s is missing", currentKCMCName), diffString
```

Comment:
Maybe it is not a good idea to error out here? Because it can just be that KubletConfig is not yet fully rendered?
Could you put the error in `diffString`? Thanks

Reply:
This is following the same logic when the kubelet.conf content is empty.

https://github.com/ComplianceAsCode/compliance-operator/pull/58/files#diff-dcf83ae774b84f0efb1bb9e9557214ab3d19130b89db9ea1107f60c084e002c9L161